<!-- IMU message attitude to TF relay (with the option to publish odom messages as well
     Warning: Our IMU's typpicaly cannot estimate translation (except for the fancy MTi-G 
     and similar with GPS. This relay fills the translation fields with zeros.
-->
<launch>

  <node pkg="imu_complementary_filter" type="complementary_filter_node"
        name="complementary_filter_gain_node" output="screen">

    <param name="do_bias_estimation" value="true"/>
    <param name="do_adaptive_gain" value="true"/>
    <param name="use_mag" value="false"/>
    <param name="gain_acc" value="0.01"/>
    <param name="gain_mag" value="0.01"/>
    <param name="publish_debug_topics" value="false"/>
    <param name="publish_tf" value="false"/>

    <remap from="/imu/data_raw" to="/mti/sensor/imu"/>
    <remap from="/imu/data" to="/mti/sensor/imu_no_mag"/>

  </node>


  <node pkg="norlab_imu_tools" type="gps_gyro_odom_to_tf_node" name="gps_gyro_odom_to_tf_node" output="screen">
    <remap from="imu_topic" to="/mti/sensor/imu" />
    <remap from="imu_nomag_topic" to="/mti/sensor/imu_no_mag" />
    <remap from="gps_odom_topic" to="/odom_utm" />
    <param name="odom_frame" value="odom" />
    <param name="base_frame" value="base_link" />

    ## This rosparam allows you to set IMU alignment w.r.t. base_link
    ## E.g. in our peli-case, the IMU is mounted upside-down, so set the RPY appropriately
    ## Values in RAD 
    <rosparam param="imu_alignment_rpy">[3.14159265359, 0.0, 3.14159265359]</rosparam>  <!--Pelicase box for skidoo, all angles in radians! -->
    <param name="mag_north_correction_yaw" value="0.4" /> <!-- 15.66 deg == 15Â°40'-->


    ## If odom message needed (but keep in mind that only the orientation makes sense):
    <param name="publish_odom" value="False" />
    <param name="odom_topic_name" value="imu_odom" />

  </node>


  <node pkg="gps_common" type="utm_odometry_node" name="gps_to_umt_odom_node" output="screen">
    <remap from="odom" to="odom_utm" />
    <param name="frame_id" value="gps_world" />
    <param name="child_frame_id" value="gps_base_link" />

  </node>

</launch>
